### ISSUE TITLE
GeneralizedRCNN - Return detections in training mode

### ISSUE THREAD

**Issue by e-dorigatti** [pytorch/vision#4749]
Currently the forward method returns only losses in training mode and only detections in eval mode. It would be useful to always return both regardless of the mode:

 1. Losses in eval mode are useful to detect overfitting
 2. Detections in training mode are useful to extend the model to perform additional learning tasks
 
## Connection to other issues
The two issues #1574 and #1775 mention returning the losses in eval mode and PR #1605 tries to implement this. I have not found anything regarding detections during training. Maybe it would be a good idea to do this in #1605 since I assume the required modifications are very similar.
